analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -0,0 +1,290 @@
+            analyzerRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) =
+            Cargo(managerName, analyzerRoot, analyzerConfig, repoConfig)
+    }
+
+    companion object {
+        private const val REQUIRED_CARGO_VERSION = "1.0.0"
+        private const val SCOPE_NAME_DEPENDENCIES = "dependencies"
+        private const val SCOPE_NAME_DEV_DEPENDENCIES = "devDependencies"
+    }
+
+    override fun command(workingDir: File?) = "cargo"
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildStrict(REQUIRED_CARGO_VERSION)
+
+    private fun runMetadata(workingDir: File): String {
+        return run(workingDir, "metadata", "--format-version=1").stdout
+    }
+
+    private fun extractCargoId(node: JsonNode): String {
+        return node["id"].textValue()!!
+    }
+
+    private fun extractPackageId(node: JsonNode) = Identifier(
+        type = "Cargo",
+        namespace = "",
+        name = node["name"].textValueOrEmpty(),
+        version = node["version"].textValueOrEmpty()
+    )
+
+    private fun extractRepositoryUrl(node: JsonNode) =
+        node["repository"]?.textValue()
+
+    private fun extractVcsInfo(node: JsonNode): VcsInfo {
+        val url = extractRepositoryUrl(node)
+        val type = if (url != null) "git" else "" // for now cargo supports only git
+        return VcsInfo(type, url.orEmpty(), revision = "")
+    }
+
+    private fun extractDeclaredLicenses(node: JsonNode): SortedSet<String> {
+        return node["license"].textValueOrEmpty().split("/")
+            .map { it.trim() }
+            .filter { it.isNotBlank() }
+            .toSortedSet()
+    }
+
+    private fun extractSourceArtifact(
+        node: JsonNode,
+        hashes: Map<String, String>
+    ): RemoteArtifact? {
+        if (node["source"].textValueOrEmpty()
+            == "registry+https://github.com/rust-lang/crates.io-index"
+        ) {
+            val name = node["name"]?.asText() ?: return null
+            val version = node["version"]?.asText() ?: return null
+            val url = "https://crates.io/api/v1/crates/$name/$version/download"
+            val checksum = checksumKeyOf(node)
+            val hash = Hash.create(hashes[checksum] ?: "")
+            return RemoteArtifact(url, hash)
+        } else {
+            return null
+        }
+    }
+
+    private fun extractPackage(node: JsonNode, hashes: Map<String, String>): Package {
+        val vcsInfo = extractVcsInfo(node)
+        return Package(
+            id = extractPackageId(node),
+            declaredLicenses = extractDeclaredLicenses(node),
+            description = node["description"].textValueOrEmpty(),
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = extractSourceArtifact(node, hashes) ?: RemoteArtifact.EMPTY,
+            homepageUrl = "",
+            vcs = vcsInfo,
+            vcsProcessed = vcsInfo.normalize()
```

Comment:
That's the default value of `vcsProcessed` and thus can be omitted. Consequently, the `vcsInfo` variable can be inlined, and this function be made an expression function.

Reply:
Ah, nice!